test/extended/node/minimum_kubelet_version.go

```diff
@@ -0,0 +1,273 @@
+	)
+})
+
+type minimumKubeletVersionAuthTestCase struct {
+	testName       string
+	kubeletVersion string
+	testFunc       func()
+}
+
+var _ = g.Describe("[sig-node][OCPFeatureGate:MinimumKubeletVersion] [Serial]", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		nodeImpersonatingClient clientset.Interface
+		nodeName                = "fakenode"
+		asUser                  = nodeNamePrefix + nodeName
+		minimumVersion          = "1.30.0"
+		f                       = framework.NewDefaultFramework("minimum-kubelet-version")
+		oc                      = exutil.NewCLIWithoutNamespace("minimum-kubelet-version")
+	)
+
+	g.BeforeEach(func() {
+		g.DeferCleanup(updateMinimumKubeletVersionAndWait(oc, minimumVersion))
+
+		ginkgo.By("Creating a kubernetes client that impersonates a node")
+		config, err := framework.LoadConfig()
+		framework.ExpectNoError(err, "failed to load kubernetes client config")
+		config.Impersonate = restclient.ImpersonationConfig{
+			UserName: asUser,
+			Groups:   []string{nodesGroup},
+		}
+		nodeImpersonatingClient, err = clientset.NewForConfig(config)
+		framework.ExpectNoError(err, "failed to create Clientset for the given config: %+v", *config)
+	})
+
+	g.It("authorization", func() {
+		testCases := []minimumKubeletVersionAuthTestCase{
+			{
+				testName:       "should be able to list pods if new enough",
+				kubeletVersion: "v1.30.0",
+				testFunc: func() {
+					_, err := nodeImpersonatingClient.CoreV1().Pods(f.Namespace.Name).List(context.Background(), metav1.ListOptions{
+						FieldSelector: "spec.nodeName=" + nodeName,
+					})
+					o.Expect(err).NotTo(o.HaveOccurred())
+				},
+			},
+			{
+				testName:       "should be able to get node",
+				kubeletVersion: "v1.29.0",
+				testFunc: func() {
+					_, err := nodeImpersonatingClient.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+				},
+			},
+			{
+				testName:       "should be able to perform subjectaccessreviews",
+				kubeletVersion: "v1.29.0",
+				testFunc: func() {
+					sar := &authorizationv1.SubjectAccessReview{
+						Spec: authorizationv1.SubjectAccessReviewSpec{
+							ResourceAttributes: &authorizationv1.ResourceAttributes{
+								Verb:      "list",
+								Resource:  "configmaps",
+								Namespace: f.Namespace.Name,
+								Version:   "v1",
+							},
+							User:   asUser,
+							Groups: []string{nodesGroup},
+						},
+					}
+
+					_, err := nodeImpersonatingClient.AuthorizationV1().SubjectAccessReviews().Create(context.Background(), sar, metav1.CreateOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+				},
+			},
+			{
+				testName:       "should block node from listing pods if too old",
+				kubeletVersion: "v1.29.0",
+				testFunc: func() {
+					_, err := nodeImpersonatingClient.CoreV1().Pods(f.Namespace.Name).List(context.Background(), metav1.ListOptions{})
+					o.Expect(err).To(o.HaveOccurred())
```

Comment:
worth checking if we received the expected error.

Reply:
good idea!